jobwatcher/jobwatcher.py

```diff
@@ -45,96 +45,128 @@ def _read_cfnconfig():
     return cfnconfig_params
 
 
-def _get_vcpus_from_pricing_file(instance_type):
+@retry(stop_max_attempt_number=3, wait_fixed=5000)
+def _get_vcpus_from_pricing_file(config):
     """
     Read pricing file and get number of vcpus for the given instance type.
 
     :param instance_type: the instance type to search for.
     :return: the number of vcpus or -1 if the instance type cannot be found
     """
-    pricing_file = "/opt/parallelcluster/instances.json"
-    with open(pricing_file) as f:
-        instances = json.load(f)
-        try:
-            vcpus = int(instances[instance_type]["vcpus"])
-            log.info("Instance %s has %s vcpus." % (instance_type, vcpus))
-        except KeyError:
-            log.error("Unable to get vcpus from file %s. Instance type %s not found." % (pricing_file, instance_type))
-            vcpus = -1
+    _create_data_dir(config.pcluster_dir)
 
-        return vcpus
+    pricing_file = config.pcluster_dir + "instances.json"
+    _fetch_pricing_file(pricing_file, config.region, config.proxy_config)
 
+    return _get_vcpus_by_instance_type(pricing_file, config.instance_type)
 
-def _get_instance_properties(instance_type):
+
+def _get_instance_properties(config):
     """
     Get instance properties for the given instance type, according to the cfn_scheduler_slots configuration parameter.
 
-    :param instance_type: instance type to search for
+    :param config: instance type to search for
```

Comment:
update description or you could leave it as it was with instance_type since that is the only config value needed.

Reply:
fixed